webpack/scenes/ContentViews/Details/Versions/BulkDelete/Steps/ConfirmBulkDelete.js

```diff
@@ -0,0 +1,103 @@
+import React, { useContext } from 'react';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { first } from 'lodash';
+import { FormattedMessage } from 'react-intl';
+import { useSelector } from 'react-redux';
+import {
+  Flex,
+  FlexItem,
+  Label,
+} from '@patternfly/react-core';
+import WizardHeader from '../../../../components/WizardHeader';
+import { selectContentViews } from '../../../../ContentViewSelectors';
+import ActionSummary from '../ActionSummary';
+import { BulkDeleteContext } from '../BulkDeleteContextWrapper';
+import {
+  getEnvironmentList,
+  getNumberOfActivationKeys,
+  getNumberOfEnviroments,
+  getNumberOfHosts,
+  getVersionListString,
+} from '../BulkDeleteHelpers';
+
+export default () => {
+  const {
+    versions, selectedCVForHosts, selectedEnvForHosts, selectedCVForAK, selectedEnvForAK,
+  } = useContext(BulkDeleteContext);
+  const { results: contentViewResults = [] } = useSelector(selectContentViews);
+  const versionList = getVersionListString(versions);
+  const numberOfActivationKeys = getNumberOfActivationKeys(versions);
+  const numberOfHosts = getNumberOfHosts(versions);
+  const affectedVersions = versions.filter(({ environments }) => !!environments.length);
+  const affectedVersionsListString = getVersionListString(affectedVersions);
+  const environments = getEnvironmentList(versions);
+
+  const pluralEnvironments = getNumberOfEnviroments(versions) > 1;
+  return (
+    <>
+      <WizardHeader
+        title={__('Review Details')}
+      />
+      <ActionSummary
+        title={__('Versions')}
+        text={<FormattedMessage
+          id="bulk-delete-summary-versions"
+          values={{
+            versionList,
+            versionOrVersions: versions.length > 1 ?
+              __('Versions') : __('Version'),
+          }}
+          defaultMessage={__('{versionOrVersions} {versionList} will be deleted and will no longer be available for promotion.')}
+        />}
+      />
+      {!!affectedVersions.length &&
+        <div>
+          <ActionSummary
+            title={__('Environments')}
+            text={
+              <FormattedMessage
+                id="bulk-delete-summary-environments"
+                values={{ versionList: affectedVersionsListString, versionOrVersions: affectedVersions.length > 1 ? __('Versions ') : __('Version ') }}
+                defaultMessage={pluralEnvironments ?
+                  __('{versionOrVersions} {versionList} will be removed from the following environments:') :
+                  __('{versionOrVersions} {versionList} will be removed from the following environment:')}
```

Comment:
Since we already have the special case for singular anyway, might be nice to say
```
__('{versionOrVersions} {versionList} will be removed from the {envLabel} environment:')
```
and then have the `envLabel` value evaluate to the pretty environment label.

Reply:
I do remember Maria saying she preferred this.. will do.. 😿